pallets/confidential-asset/src/lib.rs

```diff
@@ -34,21 +49,112 @@ pub trait Trait:
     + pallet_contracts::Trait
     + pallet_portfolio::Trait
 {
+    type Event: From<Event> + Into<<Self as frame_system::Trait>::Event>;
+}
+
+/// Wrapper for Ciphertexts that correspond to EncryptedAssetId.
+/// This is needed since `mercat::asset_proofs::elgamal_encryption::CipherText` implements
+/// Encode and Decode, instead of deriving them. As a result, the EncodeLike operator is
+/// not automatically implemented.
+#[derive(Encode, Decode, Clone, Debug, PartialEq, Eq, VecU8StrongTyped, Default)]
+pub struct EncryptedAssetIdWrapper(pub Vec<u8>);
+
+/// Wrapper for Ciphertexts that correspond to EncryptedBalance.
+/// This is needed since `mercat::asset_proofs::elgamal_encryption::CipherText` implements
+/// Encode and Decode, instead of deriving them. As a result, the EncodeLike operator is
+/// not automatically implemented.
+#[derive(Encode, Decode, Clone, Debug, PartialEq, Eq, VecU8StrongTyped, Default)]
+pub struct EncryptedBalanceWrapper(pub Vec<u8>);
+
+/// A mercat account consists of the public key that is used for encryption purposes and the
+/// encrypted asset id. The encrypted asset id also acts as the unique identifier of this
+/// struct.
+#[derive(Encode, Decode, Clone, Debug, PartialEq, Eq, Default)]
+pub struct MercatAccount {
+    pub encrypted_asset_id: EncryptedAssetIdWrapper,
+    pub encryption_pub_key: EncryptionPubKey,
 }
 
 decl_storage! {
-    trait Store for Module<T: Trait> as Asset {
+    trait Store for Module<T: Trait> as ConfidentialAsset {
+
+        /// Contains the mercat accounts for an identity.
+        pub MercatAccounts get(fn mercat_account): double_map hasher(twox_64_concat) IdentityId, hasher(blake2_128_concat) EncryptedAssetIdWrapper  => MercatAccount;
```

Comment:
Can you please try keeping to 100-character wide lines? Unfortunately some manual formatting would be required for that because `rustfmt` doesn't work with DSL macros.